src/components/App/index.tsx

```diff
@@ -78,6 +80,7 @@ const App: React.FC = ({ children }) => {
   // Load the Safes from LS just once,
   // they'll be reloaded on network change
   useEffect(() => {
+    setChainId(_getChainId())
     dispatch(loadSafesFromStorage())
     dispatch(loadCurrentSessionFromStorage())
```

Comment:
These two can be removed, as they are called from the middleware.

Reply:
Also, since this effect isn't really needed for the App component itself, I would maybe move it all to the Root component. As a place where everything in the app is initialized (remote config, chainId, etc)